src/transporters.ts

```diff
@@ -74,11 +65,25 @@ export class DefaultTransporter {
    * @param {Function=} callback Optional callback.
    * @return {Request} Request object
    */
-  request(opts: request.Options, callback?: BodyResponseCallback) {
+  request(opts: AxiosRequestConfig): AxiosPromise;
+  request(opts: AxiosRequestConfig, callback?: BodyResponseCallback): void;
+  request(opts: AxiosRequestConfig, callback?: BodyResponseCallback):
```

Comment:
Can this be an async function?

Reply:
Sorta not-really yes.  `AxiosPromise` is a Promise object, so callers of the API can just `await` it like you would a normal async function.  The reason it's not marked as `async` is because you can't have some overloaded method signatures async, and others non-async.  I want to support the mostly backward compatibility here, so that's why I opted for just returning a Promise.  To end users - it should make no difference (I think).